src/view/payment-mobile.js

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { View, StyleSheet } from 'react-native';
+import { observer } from 'mobx-react';
+import PropTypes from 'prop-types';
+import Background from '../component/background';
+import MainContent from '../component/main-content';
+import { InputField } from '../component/field';
+import { Header, Title } from '../component/header';
+import { CancelButton, Button, GlasButton } from '../component/button';
+import Card from '../component/card';
+import LightningBoltIcon from '../asset/icon/lightning-bolt';
+import { FormStretcher, FormText, FormSubText } from '../component/form';
+import QRCodeScanner from '../component/qrcode-scanner';
+import { color } from '../component/style';
+
+const styles = StyleSheet.create({
+  btnWrapper: {
+    alignSelf: 'stretch',
+    flexDirection: 'row',
+  },
+  btnLeft: {
+    flex: 1,
+    borderRightWidth: 0.5,
+    borderRightColor: color.purple,
+  },
+  btnRight: {
+    flex: 1,
+    borderLeftWidth: 0.5,
+    borderLeftColor: color.purple,
+  },
+});
+
+const PaymentView = ({ store, nav, payment }) => (
+  <Background color={color.purple}>
+    <Header color={color.purple}>
+      <Button disabled onPress={() => {}} />
+      <Title title="Lightning Payment">
+        <LightningBoltIcon height={12} width={6.1} />
+      </Title>
+      <CancelButton onPress={() => nav.goHome()} />
+    </Header>
+    {store.payment.useScanner ? (
+      <QRCodeScanner
+        onQRCodeScanned={({ data }) => payment.readQRCode({ data })}
+      />
+    ) : (
+      <PaymentCard store={store} payment={payment} />
+    )}
+    <View style={styles.btnWrapper}>
+      <GlasButton onPress={() => payment.pasteAddress()} style={styles.btnLeft}>
+        Paste
+      </GlasButton>
+      <GlasButton
+        onPress={() => payment.toggleScanner()}
+        style={styles.btnRight}
+      >
+        Scan
+      </GlasButton>
+    </View>
+  </Background>
+);
+
+PaymentView.propTypes = {
+  store: PropTypes.object.isRequired,
+  nav: PropTypes.object.isRequired,
+  payment: PropTypes.object.isRequired,
+};
+
+//
+// Payment Card
+//
+
+const cardStyles = StyleSheet.create({
+  subText: {
+    paddingTop: 15,
+    paddingLeft: 20,
+    paddingRight: 20,
+  },
+});
+
+const PaymentCard = ({ store, payment }) => (
+  <MainContent>
+    <Card>
+      <FormText>
+        Paste the Lightning Payment Request or the Bitcoin Address to which
+        you’re sending.
+      </FormText>
+      <FormStretcher>
+        <InputField
```

Comment:
Input field is too wide on the 8+: https://imgur.com/a/LvvkTLZ

Fixed by [this commit](https://github.com/lightninglabs/lightning-app/pull/821/commits/04a474e2e70b2c72030ff872fb494bd683f95b72) from my invoice PR, so maybe we can disregard for now.